nautobot/extras/api/serializers.py

```diff
@@ -795,23 +789,17 @@ class Meta:
 class SecretsGroupSerializer(NautobotModelSerializer):
     """Serializer for `SecretsGroup` objects."""
 
-    # TODO: it would be **awesome** if we could create/update SecretsGroupAssociations
-    # alongside creating/updating the base SecretsGroup, but since this is a ManyToManyField with
-    # a `through` table, that appears very non-trivial to implement. For now we have this as a
-    # read-only field; to create/update SecretsGroupAssociations you must make separate calls to the
-    # api/extras/secrets-group-associations/ REST endpoint as appropriate.
-    secrets = serializers.SerializerMethodField(read_only=True)
-
-    @extend_schema_field(SecretsGroupAssociationSerializer)
-    def get_secrets(self, obj):
-        depth = get_nested_serializer_depth(self)
-        return return_nested_serializer_data_based_on_depth(
-            self, depth, obj, obj.secrets_group_associations, "secrets_group_associations"
-        )
-
     class Meta:
         model = SecretsGroup
         fields = "__all__"
+        # TODO: it would be **awesome** if we could create/update SecretsGroupAssociations
+        # alongside creating/updating the base SecretsGroup, but since this is a ManyToManyField with
+        # a `through` table, that appears very non-trivial to implement. For now we have this as a
+        # read-only field; to create/update SecretsGroupAssociations you must make separate calls to the
+        # api/extras/secrets-group-associations/ REST endpoint as appropriate.
+        extra_kwargs = {
+            "secrets": {"source": "secrets_group_associations", "read_only": True},
+        }
```

Comment:
WOO I didn't know you can do that. Can we do something similar for `DynamicGroupSerializer`'s `children` field? And what does the field look like now?

Reply:
It **seems** to render the same way on output now. I'll take a look at DynamicGroupSerializer as well, thanks!

Reply:
<img width="1043" alt="Screen Shot 2023-05-11 at 4 18 08 PM" src="https://github.com/nautobot/nautobot/assets/46973263/68fdb968-7910-4738-87b6-0de70bd1f35b">
This is the culprit I guess.

Reply:
Weird, that just makes me wonder why in my case it's *not* using the DG URL?

Reply:
I think if you leave it up to DRF to render that field, it will use `DGMembershipSerializer` from `BaseModelSerializer` which in turn uses `NautobotHyperlinkedRelatedField`. And `NautobotHyperlinkedRelatedField` uses `get_route_for_model` instead of `get_absolute_url` from `DynamicGroupMemebrship` model class. And That gives us the correct URL.